api/redhatopenshift/HcpCluster/hcpCluster-models.tsp

```diff
@@ -440,6 +440,40 @@ model NodePoolProperties {
   spec: NodePoolSpec;
 }
 
+/** taintKey is the k8s valid key of the taint type on the nodepool nodes
+ * The good example of the taint key is `node-role.kubernetes.io/master`
+ */
+@minLength(1)
+@maxLength(316)
+scalar taintKey extends string;
+
+/** taintValue is the k8s valid value of the taint type on the nodepool nodes
+ * The good example of the taint value is `NoSchedule`
+ */
+@minLength(1)
+@maxLength(63)
+scalar taintValue extends string;
+
+/** labelValue is the k8s valid value of the label on the nodepool nodes
+ * The good example of the label value is `master`
+ */
+@minLength(1)
+@maxLength(63)
+scalar labelValue extends string;
```

Comment:
Will we need a regex restriction here? https://kubernetes.io/docs/reference/kubectl/generated/kubectl_taint/

* The key must begin with a letter or number, and may contain letters, numbers, hyphens, dots, and underscores, up to 253 characters.
* Optionally, the key can begin with a DNS subdomain prefix and a single '/', like example.com/my-app.
* The value is optional. If given, it must begin with a letter or number, and may contain letters, numbers, hyphens, dots, and underscores, up to 63 characters.

Reply:
We talked about this with @miguelsorianod. We will validate later on the same as is done internally in k8s. Originally I have written the regex, but covering all edge cases made that regex non maintainable.

Reply:
Thanks, I'm ok with that - how do we normally keep track of that, a TODO comment or follow up card?